### ISSUE TITLE
“Clean style” for dates?

### ISSUE THREAD

**Issue by nickbart1980** [plk/biblatex#446]
Is there an option to have dates formatted UK style, i.e., “day month year”, but without any ordinal indicators – a.k.a. “clean style”?

See, e.g.:

“Dates should be given in the form ‘23 April 1564’. The name of the month should always appear in full between the day (‘23’ *not* ‘23rd’) and the year.” (Modern Humanities Research Association 2013, 46)

Modern Humanities Research Association. 2013. *MHRA Style Guide: A Handbook for Authors and Editors.* 3rd ed. London: Modern Humanities Research Association. http://www.mhra.org.uk/Publications/Books/StyleGuide/.

**Comment by plk** [comment]
Currently I don't think so but I suspect it would be easy to do. Do have a MWE where they ordinals are present?

**Comment by nickbart1980** [comment]
Here’s the MWE.

~~~
\documentclass{article}
\usepackage[UKenglish]{babel}
\usepackage[autostyle]{csquotes}
\usepackage[style=authoryear, alldates=comp, dateera=christian, datezeros=false, dateeraauto=1000]{biblatex}
\usepackage{fontspec}
\setmainfont{Linux Libertine}
\usepackage{filecontents}
\begin{filecontents}{\jobname.bib}
@online{      jc,
  author    = {Julius Caesar},
  date      = {-0099-07-11/-0043-03-13},
  title     = {My Life},
  url       = {http://gov.rome},
  urldate   = {0123-07-31/0123-08-01},
}
\end{filecontents}
\addbibresource{\jobname.bib}
\begin{document}

\section*{Text}

Blah \autocite{jc}.

\printbibliography

\end{document}
~~~

Output:

![screen shot 2016-06-26 at 08 49 34](https://cloud.githubusercontent.com/assets/2765811/16361126/e00b77c4-3b73-11e6-9f23-b4ed7754ab96.png)

Expected: “Caesar, Julius (11 July 100 BC–13 Mar. 44 BC). *My Life.* URL: http://gov.rome (visited on 31 July–1 Aug. 123 AD).”

Two other issues:

- “31st July AD” has “AD” without a year.
	- Expected: “31 July–1 Aug. 123 AD”
- For ranges formatted “comp” or “terse” where start and end date share the same “CE” or “BCE” label, rendering “CE” or “BCE” only once would seem to be sufficient, e.g.:
	- “Blah (Caesar 100–44 BC).”
	- “Caesar, Julius (11th July 100–13th Mar. 44 BC).”
	- Examples from Mosshammer (2008):
		- “This is a page from St Bede’s 532-year continuation of the Easter table of Dionysius Exiguus, showing the data for the years AD 754–78.” (vi)
		- “Part Three is devoted to the history of these Alexandrian calculations between about AD 200 and 640, […]” (106)
		- “[…] and agrees with the dates given by Eusebius for the episcopate of Demetrius, AD 189/90–231/2.” (111)
		- “P covers five 19-year cycles numbered by the years of Diocletian from 145 to 239, corresponding to AD 429 to 523.” (434)

References

Mosshammer, Alden A. 2008. *The Easter Computus and the Origins of the Christian Era.* Oxford: Oxford Univ. Press.

**Comment by nickbart1980** [comment]
Ok, but BCE–CE and CE–CE dates don’t look good yet:

~~~
\documentclass{article}
\usepackage[UKenglish]{babel}
\usepackage[autostyle]{csquotes}
\usepackage[style=authoryear, alldates=comp, dateera=christian, datezeros=false, dateeraauto=1000]{biblatex}
\usepackage{fontspec}
\setmainfont{Linux Libertine}
\usepackage{filecontents}
\begin{filecontents}{\jobname.bib}
@online{      jc,
  author    = {Julius Caesar},
  date      = {-0099-07-11/-0043-03-13},
  title     = {My Life},
  url       = {http://gov.rome},
  urldate   = {0123-07-31/0123-08-01},
}
@online{      aug,
  author    = {Augustus},
  date      = {-0062-09-21/0014-08-17},
  title     = {My Life: 23 Sep 63 BCE – 19 Aug 14 CE (Julian)},
}
@online{      cal,
  author    = {Caligula},
  date      = {0012-08-29/0041-01-22},
  title     = {My Life},
}
\end{filecontents}
\addbibresource{\jobname.bib}
\DefineBibliographyExtras{UKenglish}{%
  \renewcommand*{\mkbibordinal}[1]{#1}%
}
\begin{document}
\section*{Text}
Blah \autocite{jc, aug, cal}.
\printbibliography
\end{document}
~~~

Output:

![](https://cloud.githubusercontent.com/assets/2765811/16363662/3da74320-3bc1-11e6-9438-f56647a0d6ac.png)

Expected:

“Augustus 63 BC–14 AD; Caligula 12–41 AD”

And wouldn’t the “comp” and “terse” logic also call for

“11 July 100–13 Mar. 44 BC” (rather than “11 July 100 BC–13 Mar. 44 BC”)

and

“29 Aug. 12–22 Jan. 41 AD” (rather than “29 Aug. 12 AD–22 Jan. 41 AD”)?

**Comment by plk** [comment]
Fixed - you'll need the newly built biber 2.6 too - this now outputs eras for all dates which simplifies things in biblatex.
